stdlib/public/TensorFlow/Tensor.swift

```diff
@@ -336,8 +336,64 @@ public extension Tensor {
   }
 }
 
+public extension Tensor where Scalar : BinaryInteger {
```

Comment:
This entire extension is not the right direction here for two reasons: 
- The goal of this effort is to change the **index** type to `Int`, not the type of tensor contents (scalars) or content initializers. This extension instead offers initializers that take content scalars, which are used for initializing a tensor's contents and have have nothing to do with indices.
- The naming of these newly added APIs does not conform to the Swift API Design Guidelines. Since the conversion from `Int` to arbitrary integer types is not a value-preserving conversion, the possible truncation should be mentioned in argument labels at a minimum (e.g. `Int.init(truncatingIfNeeded:)`).

In any case, this change completely orthogonal to the original goal which is to change the type of **indices** from `Int32` to `Int`. If a "truncating-if-needed" version of tensor content initialization APIs are desired in the future, they should be part of their own proposal or pull request instead of being confused with the change to indices.

Reply:
Good points. I removed this extension (for now) and replaced with manual casts from `Int` to `Int32`.

Ready for re-review.